nodejs/awsx/autoscaling/policy.ts

```diff
@@ -0,0 +1,257 @@
+}
+
+export interface PolicyArgs {
+    /**
+     * Specifies whether the adjustment is an absolute number or a percentage of the current
+     * capacity.
+     */
+    adjustmentType?: pulumi.Input<AdjustmentType>;
+
+    /**
+     * The estimated time, in seconds, until a newly launched instance will contribute CloudWatch
+     * metrics. Without a value, AWS will default to the group's specified cooldown period.
+     */
+    estimatedInstanceWarmup?: pulumi.Input<number>;
+
+    /**
+     * The minimum number of instances to scale. If the value of [adjustmentType] is
+     * ["PercentChangeInCapacity"], the scaling policy changes the DesiredCapacity of the Auto
+     * Scaling group by at least this many instances.
+     */
+    minAdjustmentMagnitude?: pulumi.Input<number>;
+}
+
+export interface SimplePolicyArgs extends PolicyArgs {
+    /**
+     * The amount of time, in seconds, after a scaling activity completes and before the next
+     * scaling activity can start.
+     */
+    cooldown?: pulumi.Input<number>;
+
+    /**
+     * The number of instances by which to scale. adjustmentType determines the interpretation of
+     * this number (e.g., as an absolute number or as a percentage of the existing Auto Scaling
+     * group size). A positive increment adds to the current capacity and a negative value removes
+     * from the current capacity.
+     */
+    scalingAdjustment?: pulumi.Input<number>;
+}
+
+export interface StepPolicyArgs extends PolicyArgs {
+    /**
+     * The aggregation type for the policy's metrics. Without a value, AWS will treat the
+     * aggregation type as "Average"
+     */
+    metricAggregationType?: pulumi.Input<MetricAggregationType>;
+
+    /**
+     * A set of adjustments that manage group scaling.
+     */
+    stepAdjustments?: pulumi.Input<pulumi.Input<StepAdjustment>[]>;
+}
+
+/** @internal */
+interface AwsTargetTrackingPolicyArgs extends PolicyArgs {
+    /**
+     * A target tracking policy.
+     */
+    targetTrackingConfiguration: aws.autoscaling.PolicyArgs["targetTrackingConfiguration"];
+}
+
+export interface TargetTrackingPolicyArgs extends PolicyArgs {
+    /**
+     * Indicates whether scaling in by the target tracking scaling policy is disabled. If scaling in
+     * is disabled, the target tracking scaling policy doesn't remove instances from the Auto
+     * Scaling group. Otherwise, the target tracking scaling policy can remove instances from the
+     * Auto Scaling group.  Defaults to [false] if unspecified.
+     */
+    disableScaleIn?: pulumi.Input<boolean>;
+
+    /**
+     * The target value for the metric.
+     */
+    targetValue: pulumi.Input<number>;
+}
+
+export interface ApplicationTargetGroupTrackingPolicyArgs extends TargetTrackingPolicyArgs {
+    /**
+     * The target group to scale [AutoScalingGroup] in response to number of requests to.
+     * If not provided, the first [TargetGroup] in [AutoScalingGroup.targetGroups] will be used.

```

Comment:
This seems like a risky default. For example, what if it's working as expected and then something gets added to the `AutoScalingGroup.targetGroups` and now the thing doesn't scale as expected anymore. I'm guessing the hope is they catch this in a diff but still seems like something we should require the user specify.

Reply:
Yes.  i see your point.  I'll make this required.